src/SystemLink.cmake

```diff
@@ -10,15 +10,23 @@ function(target_include_system_directories target)
 
   foreach(scope IN ITEMS INTERFACE PUBLIC PRIVATE)
     foreach(lib_include_dirs IN LISTS ARG_${scope})
-      if(MSVC)
+      if(NOT MSVC)
         # system includes do not work in MSVC
         # awaiting https://gitlab.kitware.com/cmake/cmake/-/issues/18272#
         # awaiting https://gitlab.kitware.com/cmake/cmake/-/issues/17904
-        target_include_directories(${target} ${scope} ${lib_include_dirs})
+        set(_SYSTEM SYSTEM)
+      endif()
+      if(${scope} STREQUAL "INTERFACE" OR ${scope} STREQUAL "PUBLIC")
```

Comment:
Not sure why this bug only happens in an INTERFACE target and not PRIVATE

Reply:
I found the reason. It is because public and interface mean that when the package is installed they should be provided by the installer (not during build).